core/domain/activity_jobs_one_off.py

```diff
@@ -589,3 +589,103 @@ def map(snapshot_model):
+
+        if isinstance(
+                snapshot_model,
+                collection_models.CollectionSnapshotMetadataModel):
+            commit_log_id = 'collection-%s-%s' % (model_id, version)
+            commit_log_model = (
+                collection_models.CollectionCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                exp_models.ExplorationRightsSnapshotMetadataModel):
+            commit_log_id = 'exploration-%s-%s' % (model_id, version)
+            commit_log_model = (
+                exp_models.ExplorationCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                question_models.QuestionSnapshotMetadataModel):
+            commit_log_id = 'question-%s-%s' % (model_id, version)
+            commit_log_model = (
+                question_models.QuestionCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                skill_models.SkillSnapshotMetadataModel):
+            commit_log_id = 'skill-%s-%s' % (model_id, version)
+            commit_log_model = (
+                skill_models.SkillCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                story_models.StorySnapshotMetadataModel):
+            commit_log_id = 'story-%s-%s' % (model_id, version)
+            commit_log_model = (
+                story_models.StoryCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                topic_models.TopicSnapshotMetadataModel):
+            commit_log_id = 'topic-%s-%s' % (model_id, version)
+            commit_log_model = (
+                topic_models.TopicCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                topic_models.SubtopicPageSnapshotMetadataModel):
+            commit_log_id = 'subtopicpage-%s-%s' % (model_id, version)
+            commit_log_model = (
+                topic_models.SubtopicPageCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if isinstance(
+                snapshot_model,
+                topic_models.TopicRightsSnapshotMetadataModel):
+            commit_log_id = 'rights-%s-%s' % (model_id, version)
+            commit_log_model = (
+                topic_models.TopicCommitLogEntryModel.get_by_id(
+                    commit_log_id))
+        if commit_log_model is None:
+            yield ('MISSING COMMIT LOGS-%s' % class_name, snapshot_model.id)
+        else:
+            yield ('FOUND COMMIT LOGS-%s' % class_name, snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, values):
+        """Implements the reduce function for this job."""
+        yield (key, values)
```

Comment:
For found commit logs you can aggregate the results.

Reply:
Done.